qml/CoverPage.qml

```diff
@@ -22,9 +22,24 @@ import Sailfish.Silica 1.0
 CoverBackground {
     id: root
 
-    property bool update: status === Cover.Active || Qt.application.active
+    // This should be deprecated only use visibleChanged.
+    property bool updateCover: status === Cover.Active
+    onUpdateCoverChanged: {
+        if (updateCover) {
+            previewImage.updatePreview()
+        }
+    }
 
-    onUpdateChanged: previewImage.updatePreview()
+    // While peeking both main window and cover window are visible at the
+    // same time. Thus, we can capture from main window when cover
+    // becomes visible.
+    onVisibleChanged: {
+        if (visible) {
+            previewImage.updatePreview()
+        } else if (window.visible) {
+            previewImage.source = ""
```

Comment:
When cover is invisible and ApplicationWindow visible we can clear the preview. Thus, you see only the next preview and not glimpses of the old preview.